modules/siddhi-extensions/map/src/main/java/org/wso2/siddhi/extension/map/ToXMLFunctionExtension.java

```diff
@@ -61,6 +77,17 @@ protected Object execute(Object data) {
     }
 
     private Object getXmlFromMap(Map<Object, Object> map) {
+        return addingElements(map);
+    }
+
+    private Object getXmlFromMapWithRootElement(Map<Object, Object> map) {
+        String xmlValue = "<" + rootElement + ">";
+        xmlValue += addingElements(map);
+        xmlValue += "</" + rootElement + ">";
+        return xmlValue;
```

Comment:
Thanks for the suggestion. Fixed with https://github.com/wso2/siddhi/pull/189/commits/6ec926d984a25a5000d33f3696352a1d730628c3